contracts/beacon/Voucher.sol

```diff
@@ -0,0 +1,469 @@
+     * @param datasetOrder The dataset order.
+     * @param workerpoolOrder The workerpool order.
+     * @param requestOrder The request order.
+     */
+    function matchOrdersBoost(
+        IexecLibOrders_v5.AppOrder calldata appOrder,
+        IexecLibOrders_v5.DatasetOrder calldata datasetOrder,
+        IexecLibOrders_v5.WorkerpoolOrder calldata workerpoolOrder,
+        IexecLibOrders_v5.RequestOrder calldata requestOrder
+    ) external onlyAuthorized onlyNotExpired returns (bytes32 dealId) {
+        VoucherStorage storage $ = _getVoucherStorage();
+        IVoucherHub voucherHub = IVoucherHub($._voucherHub);
+        address iexecPoco = voucherHub.getIexecPoco();
+        uint256 sponsoredAmount = _debitVoucherAndTransferNonSponsoredAmount(
+            $._type,
+            voucherHub,
+            iexecPoco,
+            appOrder,
+            datasetOrder,
+            workerpoolOrder,
+            requestOrder
+        );
+        dealId = IexecPocoBoost(iexecPoco).sponsorMatchOrdersBoost(
+            appOrder,
+            datasetOrder,
+            workerpoolOrder,
+            requestOrder
+        );
+        $._sponsoredAmounts[dealId] = sponsoredAmount;
+        emit OrdersBoostMatchedWithVoucher(dealId);
+        return dealId;
+    }
+
+    /**
+     * Claim failed task on PoCo then refund voucher and requester.
+     * @param taskId id of the task
+     */
+    function claim(bytes32 taskId) external {
+        VoucherStorage storage $ = _getVoucherStorage();
+        IVoucherHub voucherHub = IVoucherHub($._voucherHub);
+        address iexecPoco = voucherHub.getIexecPoco();
+        IexecLibCore_v5.Task memory task = IexecPocoAccessors(iexecPoco).viewTask(taskId);
+        // Claim task on PoCo if not already claimed.
+        // This implicitly validates that the task and its deal exist.
+        if (task.status != IexecLibCore_v5.TaskStatusEnum.FAILED) {
+            IexecPoco2(iexecPoco).claim(taskId);
+        }
+        IexecLibCore_v5.Deal memory deal = IexecPocoAccessors(iexecPoco).viewDeal(task.dealid);
+        // If the deal was matched by the voucher, then the voucher should be refunded.
+        // If the deal was partially or not sponsored by the voucher, then the requester
+        // should be refunded.
+        if (deal.sponsor == address(this)) {
+            _refundVoucherAndRequester(
+                voucherHub,
+                iexecPoco,
+                taskId,
+                deal.app.price + deal.dataset.price + deal.workerpool.price, // taskPrice
+                task.dealid,
+                deal.botSize,
+                deal.requester
+            );
+        }
+        emit TaskClaimedWithVoucher(taskId);
+    }
+
+    /**
+     * Claim failed Boost task on PoCo then refund voucher and requester.
+     * @param dealId id of the task's deal
+     * @param taskIndex task's index in the deal
+     */
+    function claimBoost(bytes32 dealId, uint256 taskIndex) external {
```

Comment:
Because in Boost mode, there is not `initialize()` so there is not task definition before pushing the result of compute.
Check Boost delegate on PoCo https://github.com/iExecBlockchainComputing/PoCo/blob/develop/contracts/modules/delegates/IexecPocoBoostDelegate.sol#L371